test/page.spec.js

```diff
@@ -781,6 +781,126 @@ module.exports.addTests = function({testRunner, expect, puppeteer, DeviceDescrip
     });
   });
 
+  describe('Page.waitForRequest', function() {
+    it('should work', async({page, server}) => {
+      await page.goto(server.EMPTY_PAGE);
+
+      server.setRoute('/getRequest', (req, res) => {
```

Comment:
there's no need to do `setRoute` for server: it's fine to fetch non-existing resources (and get 404)